flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/functions/SqlLikeUtils.java

```diff
@@ -108,15 +108,18 @@ static String sqlToRegexLike(String sqlPattern, char escapeChar) {
         final StringBuilder javaPattern = new StringBuilder(len + len);
         for (i = 0; i < len; i++) {
             char c = sqlPattern.charAt(i);
-            if (JAVA_REGEX_SPECIALS.indexOf(c) >= 0) {
-                javaPattern.append('\\');
-            }
             if (c == escapeChar) {
                 if (i == (sqlPattern.length() - 1)) {
                     throw invalidEscapeSequence(sqlPattern, i);
                 }
                 char nextChar = sqlPattern.charAt(i + 1);
-                if ((nextChar == '_') || (nextChar == '%') || (nextChar == escapeChar)) {
+                if ((nextChar == '_') || (nextChar == '%')) {
```

Comment:
We'd better also add some tests into `FlinkSqlLikeUtilsTest` to cover these changes.